### PR TITLE
Fix/ftdi/usb hardware add

### PR DESCRIPTION
**Pull request by rfuentess**
### Contribution description

While working with physical boards iotlab-m3 I noticed that they were burned without a serial field, restricting RIOT ability of working with multiple of them at the same time. 

This is what RIOT sees when two iotlab-m3 boards are up:

```
~/RIOT/examples/hello-world$ make list-ttys 
/sys/bus/usb/devices/4-1.7.1.6: IoT-LAB M3 serial: '', tty(s): ttyUSB2
/sys/bus/usb/devices/4-1.7.1.5.2: HiKoB FITECO M3 serial: '', tty(s): ttyUSB1
```

`make flash` will works but will be always with `ttyUSB1`. The alternative is to use the real USB hardware address which requires the use of `ftdi_location` command instead of `ftdi_serial`. 

Also, after a off-line discussion with @kYc0o , we opted to concentrate all the current work for the FTDI driver, and this addition,  in a single point. This is why the mulle board is also impacted by this commit, as was already using the FTDI driver.

### Issues/PRs references

None

### CODE REVIEW

File: makefiles/tools/openocd-adapters/ftdi.inc.mk [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# ftdi-serial debug adapter
+# Add serial matching command, only if DEBUG_ADAPTER_ID was specified
+ifneq (,$(DEBUG_ADAPTER_ID))
+  OPENOCD_ADAPTER_INIT += -c 'ftdi_serial $(DEBUG_ADAPTER_ID)'
+else ifeq (,$(DEBUG_ADAPTER_LOCATION))
+  #Some boards doesn't have serial at all (eg. itolab-m3)
```
Comment (kYc0o) [inline makefiles/tools/openocd-adapters/ftdi.inc.mk]: Please put a space after `#` and rephrase as `Some boards don't have ...`

**Comment by gebart** [comment]
See also #7695

**Comment by rfuentess** [comment]
The work of this PR is halted because I lost the ability of flashing my available m3 boards.
I hope to be able to fix this inconvenient ASAP.

**Comment by tcschmidt** [comment]
@rfuentess flashing fixed?

**Comment by rfuentess** [comment]
Hi  @tcschmidt, I was able to recover my boards with PR #8977. But, I forgot this PR. I should be retaking the work today or tomorrow.

**Comment by rfuentess** [comment]
Finally I was able to test my  last fixups, @kYc0o, comments addressed :smile:

**Comment by tcschmidt** [comment]
@kYc0o do you confirm?

**Comment by kYc0o** [comment]
It needs rebase. Codewise (or rather make-wise) looks ok. However we need #7695 to come first anyways.

**Comment by tcschmidt** [comment]
Yep - maybe let's help @gebart a bit?

**Comment by rfuentess** [comment]
I was testing this PR under a rebase with #7695.  Without a1bae33022f1b8023520c1209bfca0a0264e7a81 they can operate with transparency. Said commit was only moving the default behavior for mulle from one file to another, while PR 7695 is eliminating said behavior.  

> It needs rebase. Codewise (or rather make-wise) looks ok. However we need #7695 to come first anyways.

 I can rebase now with @gebart's PR instead of the current master.

**Pull request closed without merging**